lib/spack/spack/util/path.py

```diff
@@ -371,6 +371,7 @@ def padding_filter(string):
     entirety at least one time. e.g., "/spack/" would not be filtered, but
     "/__spack_path_placeholder__/spack/" would be.
 
+    Note that only the first padded path in the string is filtered.
```

Comment:
Did you encounter cases where the padded path appeared multiple times in a string?  How hard would it be to filter every instance?  Not sure if it's just a matter of supplying the `count` parameter to `sub()`, or supplying a combination of flags e.g. `re.MULTILINE | re.DOTALL`?

Reply:
The comment is per-call so, IIRC, the path filtering works fine where it has been added.  For example, the `patchelf` command that was causing issues with the `gptune` log for #27877 that we discussed before shows each dependency path padded as shown below:

```
==> [2022-08-09-00:53:57.354561] '/usr/bin/patchelf' '--force-rpath' '--set-rpath' '/home/software/spack/[padded-to-512-chars]/linux-ubuntu22.04-x86_64/gcc-11.2.0/pkgconf-1.8.0-wsqf3pdlaknj3vt7bq7skjmztdd5flps/lib:/home/software/spack/[padded-to-600-chars]/morepadding/linux-ubuntu22.04-x86_64/gcc-11.2.0/pkgconf-1.8.0-wsqf3pdlaknj3vt7bq7skjmztdd5flps/lib:/home/software/spack/[padded-to-612-chars]/linux-ubuntu22.04-x86_64/gcc-11.2.0/pkgconf-1.8.0-wsqf3pdlaknj3vt7bq7skjmztdd5flps/lib64:/home/software/spack/[padded-to-602-chars]/morepadding/linux-ubuntu22.04-x86_64/gcc-11.2.0/pkgconf-1.8.0-wsqf3pdlaknj3vt7bq7skjmztdd5flps/lib64' '/home/software/spack/[padded-to-512-chars]/linux-ubuntu22.04-x86_64/gcc-11.2.0/pkgconf-1.8.0-wsqf3pdlaknj3vt7bq7skjmztdd5flps/bin/pkgconf'
```

(From https://gitlab.spack.io/spack/spack/-/jobs/2875679.)

Reply:
As for filtering each instance, the API *could* certainly be changed.  I tried to keep the modifications to a minimum for this PR though.